server/file.js

```diff
@@ -27,22 +33,30 @@ function formResponse(content, filePath) {
 
 function applyRemark(markdown) {
   return new Promise((resolve, reject) => {
-    remark().use(lint).use(html).process(markdown, (err, markdownHtml) => {
-      if (err) {
-        reject(err.message);
-      }
+    remark()
+      .use(lint)
+      .use(html)
+      .process(markdown, (err, markdownHtml) => {
+        if (err) {
+          reject(err.message);
+        }
 
-      resolve(String(markdownHtml));
-    });
+        resolve(String(markdownHtml));
+      });
   });
 }
 
-function getFileContent(filePath) {  
+function getFileContent(filePath) {
+  currentFilePath = path.resolve(filePath);
```

Comment:
there will be a flaw in this logic. If you open 2 tabs with different files, the last one that does the request overwrites this var and you don't get notifications on the other tab. Maybe, just do notify with change and filename and client keep the state and deciding, whether to show the notification?